content/blog/pulumi-kubernetes-operator-2-0/index.md

```diff
@@ -0,0 +1,269 @@
+
+```yaml
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: nginx-k8s-stack-production
+spec:
+  stack: acmecorp/nginx/production
+  projectRepo: https://github.com/acmecorp/pulumi-nginx
+  commit: 2b0889718d3e63feeb6079ccd5e4488d8601e353
+```
+
+The Pulumi Kubernetes Operator can also be used along with Pulumi’s [Kubernetes provider](/registry/packages/kubernetes/), which allows Pulumi programs to deploy resources to Kubernetes, either in the same cluster as the operator, or in another cluster. This can be used to package up complex Kubernetes infrastructure into more abstracted units, to mix Kubernetes and cloud infrastructure into a single unit of deployment, to define how application workloads are deployed into a cluster, or for a wide variety of additional use cases. The Pulumi Kubernetes provider has full support for the Kubernetes API available in every Pulumi language, including support for [Helm](/blog/full-access-to-helm-features-through-new-helm-release-resource-for-kubernetes/).
+
+## What’s New in Pulumi Kubernetes Operator 2.0?
+
+The 2.0 release is based on a whole new architecture for running Pulumi programs in your Kubernetes cluster.
+The operator now allocates a dedicated pod for each `Stack` to serve as the execution environment for Pulumi stack operations.
+Previously, all stack operations took place within the operator's own pod. This new approach effectively isolates
+each stack's compute and memory resources, improves the isolation of secrets, and opens up new customization options.
+
+These pods are referred to as *workspace pods*.  Under the hood, a new `Workspace` Custom Resource handles the lifecycle
+of the workspace pod, and is created automatically by the system. A related `Update` Custom Resource applies a stack operation to a given workspace.
+
+The `Stack` Custom Resource is still the primary API for using the Pulumi Kubernetes Operator and is largely unchanged.
+See the Migration section below to understand how to prepare your `Stack` objects for use with the 2.0 release.
+
+Note: a key limitation of the new system is that cross-namespace references are strictly forbidden. For example, a `Stack` cannot refer
+to a `Secret` in another namespace.
+
+### Installation
+
+With the 2.0 release, the supported installation modes are limited to a cluster-wide installation. You'll need to be
+a cluster admin to install the operator. The operator is able to handle stacks across all the namespaces of your cluster.
+
+We provide three ways to install the operator:
+
+1. a Pulumi program (see: `deploy/pulumi-operator-yaml/`)
+2. a Helm chart (see: `deploy/helm/pulumi-operator/`)
+3. a simple manifest file (see: `deploy/yaml/install.yaml`)
+
+Please uninstall any earlier version of the Pulumi Kubernetes Operator, then install the new version using one of the above options.
+Note that this will remove any existing `Stack` objects in your cluster, so be sure to export their manifests or have
+a way to restore them.
```

Comment:
Is there a safer route where they upgrade CRDs instead of destroying them?

Reply:
I wholeheartedly agree but the effort is beyond the scope of the beta.